cmd/client/grpc_client.go

```diff
@@ -62,7 +64,20 @@ func Conn(ctx context.Context, remote string) (*grpc.ClientConn, error) {
 
 	ctx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
-	return grpc.DialContext(ctx, remote, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock(), grpc.WithDisableHealthCheck())
+	return grpc.DialContext(ctx, remote,
+		grpc.WithTransportCredentials(transportCredentials(remote)),
+		grpc.WithBlock(),
+		grpc.WithDisableHealthCheck())
+}
+
+func transportCredentials(remote string) credentials.TransportCredentials {
+	u, err := url.Parse(remote)
+	if err == nil && (u.Hostname() == "127.0.0.1" || u.Hostname() == "localhost") {
```

Comment:
I think you rather have to use `net.SplitHostPort`, as `127.0.0.1` is likely not parsed correctly...
https://go.dev/play/p/80PCAF8EtBL